utils/yoke/src/yokeable.rs

```diff
@@ -0,0 +1,62 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::borrow::{Cow, ToOwned};
+use std::mem;
+
+pub unsafe trait Yokeable<'a>: 'static {
+    type Output: 'a + Sized; // MUST be `Self` with the lifetime swapped
+                             // used by `SharedData::get()`
+    fn transform(&'a self) -> &'a Self::Output;
+
+    // Used for zero-copy deserialization. Safety constraint: `Self` must be
+    // destroyed before the data `Self::Output` was deserialized
+    // from is
+    unsafe fn make(from: Self::Output) -> Self;
+
+    fn with_mut<F>(&'a mut self, f: F)
+    where
+        F: 'static + FnOnce(&'a mut Self::Output);
+}
+
+unsafe impl<'a, T: 'static + ToOwned + ?Sized> Yokeable<'a> for Cow<'static, T>
+where
+    <T as ToOwned>::Owned: Sized,
+{
+    type Output = Cow<'a, T>;
+    fn transform(&'a self) -> &'a Cow<'a, T> {
+        self
+    }
```

Comment:
This doesn't require unsafe because... shortening lifetimes is always safe?

Reply:
for covariant lifetimes, yes (which is a part of the `unsafe trait`'s assumptions)